src/components/victory-candlestick/helper-methods.js

```diff
@@ -0,0 +1,130 @@
+import { pick, omit, defaults, flatten } from "lodash";
+import { Helpers, Events } from "victory-core";
+import Scale from "../../helpers/scale";
+import Domain from "../../helpers/domain";
+// import Data from "../../helpers/data";
+
+export default {
+  getBaseProps(props, defaultStyles) { // eslint-disable-line max-statements
+    const calculatedValues = this.getCalculatedValues(props, defaultStyles);
+    const { data, style, scale } = calculatedValues;
+    return data.reduce((memo, datum, index) => {
+      const eventKey = datum.eventKey;
+      const x = scale.x(datum.x);
+      const y1 = scale.y(datum.high);
+      const y2 = scale.y(datum.low);
+      const candleColor = datum.open > datum.close ?
+            props.candleColors.negative : props.candleColors.positive;
+      const candleHeight = Math.abs(scale.y(datum.open) - scale.y(datum.close));
+      const y = scale.y(Math.max(datum.open, datum.close));
+      const size = this.getSize(datum, props, calculatedValues);
+      const dataStyle = Object.assign(this.getDataStyles(datum, style.data),
```

Comment:
actually cleaner to pass `props` into `getDataStyles` and have this logic happen there.